test/testDownloadManager.js

```diff
@@ -0,0 +1,291 @@
+      repo: 'test-repo',
+    }), Error('Unknown path, cannot fetch content'));
+  });
+
+  it('fetchGithub warns if no HTTP_TIMEOUT', () => {
+    const testLogger = logging.createTestLogger({
+      // tune this for debugging
+      level: 'info',
+    });
+
+    // eslint-disable-next-line no-new
+    new DownloadManager({}, {
+      secrets: {},
+      logger: testLogger,
+    });
+    assert.equal(testLogger.getOutput(), 'warn: No HTTP timeout set, risk of denial-of-service\n');
+  });
+
+  it('can suppress 404 error.', async () => {
+    nock('https://raw.githubusercontent.com')
+      .get('/adobe/test-repo/development/myfile1.md')
+      .delay(30)
+      .reply(() => [404, 'not found']);
+
+    const context = {};
+    const action = coerce({
+      logger,
+    });
+
+    const mgr = new DownloadManager(context, action);
+    const res = await mgr.fetchGithub({
+      owner: 'adobe',
+      repo: 'test-repo',
+      ref: 'development',
+      path: '/myfile1.md',
+      errorOn404: false,
+    });
+
+    assert.equal(res.ok, false);
+    assert.equal(res.status, 404);
+    assert.equal(await res.text(), 'not found');
+    assert.equal(context.error, undefined);
+  });
+
+  it('throws on 404 error if desired.', async () => {
+    nock('https://raw.githubusercontent.com')
+      .get('/adobe/test-repo/development/myfile1.md')
+      .delay(30)
+      .reply(() => [404, 'not found']);
+
+    const context = {};
+    const action = coerce({
+      logger,
+    });
+
+    const mgr = new DownloadManager(context, action);
+    try {
+      await mgr.fetchGithub({
+        owner: 'adobe',
+        repo: 'test-repo',
+        ref: 'development',
+        path: '/myfile1.md',
+        errorOn404: true,
+      });
+      assert.fail('should throw error.');
+    } catch (err) {
+      assert.equal(err.message, 'not found');
+      assert.equal(err.status, 404);
+      assert.equal(context.error.message, 'not found');
+    }
+  });
+
+  it('can schedule downloads in parallel', async () => {
+    nock('https://raw.githubusercontent.com')
+      .get('/adobe/test-repo/development/myfile1.md')
+      .delay(30)
+      .reply(() => [200, 'hello, world 1.'])
+      .get('/adobe/test-repo/development/myfile2.md')
+      .delay(20)
+      .reply(() => [200, 'hello, world 2.'])
+      .get('/adobe/test-repo/development/myfile3.md')
+      .delay(10)
+      .reply(() => [200, 'hello, world 3.']);
+
+    nock('https://www.example.com')
+      .get('/data')
+      .delay(30)
+      .reply(() => [200, 'hello, world 4.']);
+
+    const action = coerce({
+      logger,
+    });
+    const mgr = new DownloadManager({}, action);
+    ['/myfile1.md', '/myfile2.md', '/myfile3.md'].forEach((path) => {
+      mgr.fetchGithub({
+        owner: 'adobe',
+        repo: 'test-repo',
+        ref: 'development',
+        path,
+        id: path,
+      });
+    });
+
+    // todo: this doesn't seem to work. maybe nock cannot intercept this. but why with github ?
```

Comment:
@stefan-guggisberg Somehow, I can't get `nock` to intercept `www.example.com`. but it works for `raw. githubusercontent.com`. I don't understand the difference :-) I even tried to force `http1` protocol...but it just doesn't work :-)

Reply:
here's the difference ;)

https://http2.pro/check?url=https%3A//www.example.com/
https://http2.pro/check?url=https%3A//raw.githubusercontent.com/